src/averageAges.js

```diff
@@ -37,7 +38,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(female => withChildren
+    ? female.sex === 'f' && people.some(child => child.mother === female.name)
+    : female.sex === 'f'
+  );
+
+  const averageAge = (women
```

Comment:
Consider renaming `averageAge` to `totalAge` for consistency with the `calculateMenAverageAge` function.

and then return `totalAge / women.length`
it will be more readable this way